lib/active_model/serializer/pagination_serializer.rb

```diff
@@ -0,0 +1,16 @@
+module ActiveModel
+  class Serializer
+    class PaginationSerializer < ArraySerializer
+      attr_reader :page_size, :page_number
+
+      def initialize(objects, options = {})
+        @page_size   = options.delete(:page_size)
+        @page_number = options.delete(:page_number)
+
+        super
+
+        @objects = @objects.first.instance_variable_get('@objects')
```

Comment:
originally ``@objects`` referred to the ``ArraySerializer`` any thoughts on how to do this better?